File: gamedex-datamodel/src/main/java/com/github/ykrasik/gamedex/datamodel/provider/SearchResult.java
package com.github.ykrasik.gamedex.datamodel.provider;

import com.github.ykrasik.yava.option.Opt;
import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

import java.time.LocalDate;

/**
 * @author Yevgeny Krasik
 */
@Value
@Builder
public class SearchResult {
    @NonNull private final String detailUrl;
    @NonNull private final String name;
    @NonNull private final Opt<LocalDate> releaseDate;
    @NonNull private final Opt<Double> score;
}

File: gamedex-data-provider/src/main/kotlin/com/gitlab/ykrasik/gamedex/provider/SearchResult.kt
package com.gitlab.ykrasik.gamedex.provider

import org.joda.time.LocalDate

/**
 * User: ykrasik
 * Date: 29/05/2016
 * Time: 18:28
 */
data class SearchResult(
    val detailUrl: String,
    val name: String,
    val releaseDate: LocalDate?,
    val score: Double?
)